### PR TITLE
Add coordinator scaffolding

### PR DESCRIPTION
**Pull request by jferrant**

### CODE REVIEW

File: sbtc-core/src/signer/mod.rs [hunk 1]
```diff
@@ -143,31 +125,3 @@ impl<S: Sign + Coordinate> Keys for Signer<S> {
         todo!()
     }
 }
-
```
Comment (xoloki) [inline sbtc-core/src/signer/mod.rs:146]: Did you mean to remove the skeleton implementations?

**Review by xoloki — APPROVED** [review]

File: sbtc-core/src/signer/coordinator/mod.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/// Signers' public keys required for weighted distributed signing
+pub struct PublicKeys {
+    /// Signer ids to public key mapping
+    pub signer_ids: HashMap<u32, ecdsa::PublicKey>,
+    /// Vote ids to public key mapping
+    pub vote_ids: HashMap<u32, ecdsa::PublicKey>,
+}
+
+/// TODO: Define the Message types for DKG round
+/// TODO: Define the Message types for Tx Signning Round
```
Comment (netrome) [inline sbtc-core/src/signer/coordinator/mod.rs:23]: Bonus points for linking issues in the todo comments

**Review by netrome — APPROVED** [review]

File: sbtc-core/src/signer/mod.rs (hunk 1 above)
Comment (jferrant) [reply sbtc-core/src/signer/mod.rs:146]: Yes I did. The coordinator should actually be implemented on the frost-coordinator, fire-coordinator, roast-coordinator level and then passed to the Signer struct.

File: sbtc-core/src/signer/coordinator/mod.rs (hunk 2 above)
Comment (jferrant) [reply sbtc-core/src/signer/coordinator/mod.rs]: Done

**Pull request merged**